src/porepy/numerics/fv/biot.py

```diff
@@ -393,143 +388,184 @@ def _discretize_mech(self, g, data):
+            discretization of the jumps in [n alpha p] term,
+            ready to be multiplied with inverse gradient
+            scipy.sparse.csr_matrix (shape num_subfaces * dim, num_cells):
+                discretization of the force on the face due to cell-centre
+                pressure from a unique side. 
+
+        Method properties and implementation details.
+        Basis functions, namely 'stress' and 'bound_stress', for the displacement
+        discretization are obtained as in standard MPSA-W method.
+        Pressure is represented as forces in the cells.
+        However, jumps in pressure forces over a cell face act as force
+        imbalance, and thus induce additional displacement gradients in the sub-cells.
+        An additional system is set up, which applies non-zero conditions to the
+        traction continuity equation. This can be expressed as a linear system on the form
+            (i)   A * grad_u            = I
+            (ii)  B * grad_u + C * u_cc = 0
+            (iii) 0            D * u_cc = 0
+        Thus (i)-(iii) can be inverted to express the additional displacement gradients
+        due to imbalance in pressure forces as in terms of the cell center variables.
+        Thus we can compute the basis functions 'grad_p_jumps' on the sub-cells.
+        To ensure traction continuity, as soon as a convention is chosen for what side
+        the force evaluation should be considered on, an additional term, called
+        'grad_p_face', is added to the full force. This latter term represnts the force
+        due to cell-center pressure acting on the face from the chosen side.
+        The pair subfno_unique-unique_subfno gives the side convention.
+        The full force on the face is therefore given by
+        t = stress * u + bound_stress * u_b + (grad_p_jumps + grad_p_face) * p
+
+        The strategy is as follows.
+        1. compute product normal_vector * alpha and get a map for vector problems
+        2. assemble r.h.s. for the new linear system, needed for the term 'grad_p_jumps'
+        3. compute term 'grad_p_face'
+        """
+        
+        nd = g.dim
 
-        if g.dim == 2:
-            alpha_tensor.values = np.delete(alpha_tensor.values, (2), axis=0)
-            alpha_tensor.values = np.delete(alpha_tensor.values, (2), axis=1)
+        num_subhfno = subcell_topology.subhfno.size
+        num_subfno_unique = subcell_topology.num_subfno_unique
+        num_subfno = subcell_topology.num_subfno
+        num_cno = subcell_topology.num_cno
 
-        # Compute nAlpha product same as nK in Darcy
-        nAlpha_grad, cell_node_blocks, sub_cell_index = fvutils.scalar_tensor_vector_prod(
-            g, alpha_tensor, subcell_topology
-        )
+        num_nodes = np.diff(g.face_nodes.indptr)
+
+        # Step 1
+
+        # Take Biot's alpha as a tensor
+        alpha_tensor = tensor.SecondOrder(nd, alpha * np.ones(g.num_cells))
```

Comment:
This is old syntax and should not run in this branch. If it is running on your computer it is because python still has access to the old version of porepy you have on your computer. Same goes for line 483 (should be alpha_tensor.values)